test/lib/ls.js

```diff
@@ -0,0 +1,2510 @@
+const { test } = require('tap')
+const requireInject = require('require-inject')
+
+const simpleNmFixture = {
+  node_modules: {
+    'foo': {
+      'package.json': JSON.stringify({
+        name: 'foo',
+        version: '1.0.0',
+        dependencies: {
+          'bar': '^1.0.0'
+        }
+      })
+    },
+    'bar': {
+      'package.json': JSON.stringify({
+        name: 'bar',
+        version: '1.0.0'
+      })
+    },
+    'lorem': {
+      'package.json': JSON.stringify({
+        name: 'lorem',
+        version: '1.0.0'
+      })
+    }
+  }
+}
+
+const diffDepTypesNmFixture = {
+  node_modules: {
+    'dev-dep': {
+      'package.json': JSON.stringify({
+        name: 'dev-dep',
+        description: 'A DEV dep kind of dep',
+        version: '1.0.0',
+        dependencies: {
+          'foo': '^1.0.0'
+        }
+      })
+    },
+    'prod-dep': {
+      'package.json': JSON.stringify({
+        name: 'prod-dep',
+        description: 'A PROD dep kind of dep',
+        version: '1.0.0',
+        dependencies: {
+          'bar': '^2.0.0'
+        }
+      }),
+      node_modules: {
+        bar: {
+          'package.json': JSON.stringify({
+            name: 'bar',
+            description: 'A dep that bars',
+            version: '2.0.0'
+          })
+        }
+      }
+    },
+    'optional-dep': {
+      'package.json': JSON.stringify({
+        name: 'optional-dep',
+        description: 'Maybe a dep?',
+        version: '1.0.0'
+      })
+    },
+    'peer-dep': {
+      'package.json': JSON.stringify({
+        name: 'peer-dep',
+        description: 'Peer-dep description here',
+        version: '1.0.0'
+      })
+    },
+    ...simpleNmFixture.node_modules
+  }
+}
+
+let prefix
+let result = ''
+const _flatOptions = {
+  dev: false,
+  depth: Infinity,
+  json: false,
+  link: false,
+  only: null,
+  parseable: false,
+  production: false
+}
+const ls = requireInject('../../lib/ls.js', {
+  '../../lib/npm.js': {
+    flatOptions: _flatOptions,
+    limit: {
+      fetch: 3
+    },
+    get dir () { return prefix + '/node_modules/' },
+    globalDir: '/foo',
+    config: {
+      get (key) {
+        return _flatOptions[key]
+      }
+    }
+  },
+  '../../lib/utils/output.js': msg => { result = msg }
+})
+
+const redactCwd = res =>
```

Comment:
This is going to be a problem on Windows, because `\\` paths and upper/lowercase drive letters.

So, you can't trivially pass it to `new RegExp`, because the `\\` will escape whatever comes after it.  Also, even if you get the regexp to work, then you'll have `<cwd>\foo.js` instead of `<cwd>/foo.js`.

Something like this might work:

```js
res.replace(new RegExp(__dirname.replace(/\\/g, '/'), 'gi'), '<cwd>').replace(/\\/g, '/')
```

Reply:
cool 👍 will change it and rerun local tests again 😊

Reply:
while I'm on it will pb also replace `<cwd>` with `{CWD}` which was what I used in `map-workspaces` to at least try and keep a standard around those